cpp/rss_subset_aggregator.cc

```diff
@@ -0,0 +1,319 @@
+                                  + "' WHERE id=" + user_id);
+    } else
+        GenerateFeed(subsystem_type, harvested_items);
+    return true;
+}
+
+
+// Yes, this function has a confusing name but I could not think of a better one. :-(
+// What is meant is how to address a user!
+std::string GenerateUserAddress(const std::string &appellation, const std::string &first_name, const std::string &last_name) {
+    if (last_name.empty())
+        return first_name;
+
+    if (appellation.empty())
+        return first_name + " " + last_name;
+
+    return appellation + " " + last_name;
+}
+
+
+} // unnamed namespace
+
+
+struct UserInfo {
+    std::string user_id_;
+    std::string first_name_;
+    std::string last_name_;
+    std::string email_;
+    std::string rss_feed_last_notification_;
+public:
+    UserInfo() = default;
+    UserInfo(const UserInfo &other) = default;
+    UserInfo(const std::string &user_id, const std::string &first_name, const std::string &last_name,
+             const std::string &email, const std::string &rss_feed_last_notification)
+        : user_id_(user_id), first_name_(first_name), last_name_(last_name), email_(email),
+          rss_feed_last_notification_(rss_feed_last_notification) { }
+};
+
+
+int Main(int argc, char *argv[]) {
+    if (argc != 3)
+        Usage();
+
+    std::string error_email_address, vufind_user_id;
+    if (std::strcmp(argv[1], "--mode=email") == 0)
+        vufind_user_id = argv[1];
+    else if (std::strcmp(argv[1], "--mode=rss_xml") == 0)
+        vufind_user_id = argv[1];
+    else
+        Usage();
+
+    const auto db_connection(VuFind::GetDbConnection());
+
+    std::string sql_query("SELECT id,firstname,lastname,email,tuefind_rss_feed_send_emails"
+                          ",tuefind_rss_feed_last_notification FROM user");
+    if (vufind_user_id.empty())
+        sql_query += " WHERE tuefind_rss_feed_send_emails IS TRUE";
+    else
+        sql_query += " WHERE id=" + db_connection->escapeAndQuoteString(vufind_user_id);
+    db_connection->queryOrDie(sql_query);
+
+    auto user_result_set(db_connection->getLastResultSet());
+    std::unordered_map<std::string, UserInfo> ids_to_user_infos_map;
+    while (const auto user_row = user_result_set.getNextRow())
+        ids_to_user_infos_map[user_row["id"]] =
+            UserInfo(user_row["id"], user_row["firstname"], user_row["lastname"], user_row["email"],
+                     user_row["tuefind_rss_feed_last_notification"]);
+
+    unsigned feed_generation_count(0), email_sent_count(0);
+    for (const auto &[user_id, user_info] : ids_to_user_infos_map) {
+        if (vufind_user_id.empty() and not EmailSender::IsValidEmailAddress(user_info.email_)) {
+            LOG_WARNING("no valid email address for vfind.user.id " + user_id + "!");
+            continue;
+        }
+
+        db_connection->queryOrDie("SELECT appellation,language,user_type FROM ixtheo_user");
```

Comment:
You cannot use the ixtheo_user table. This functionality will also be used for KrimDok!